vllm/model_executor/layers/fused_moe/deep_gemm_moe.py

```diff
@@ -127,6 +123,14 @@ def apply(
             self.block_shape[0],
         )
 
+        if expert_map is not None:
+            # DeepGemm (Grouped Contiguous) kernel needs a valid B index
+            # for all rows of A. To that effect, simply compute with
+            # the 0th weight matrix.
+            # Note that this relies on the fact that corresponding topk
+            # weights would be 0 during weight multiplication.
+            expert_ids = torch.where(expert_ids == -1, 0, expert_ids)
```

Comment:
For clarity: In this case we're still multiplying a valid weight matrix (expert 0) with a valid token, so we shouldn't expect any `inf`s or `NaN`s in the output, right?